packages/ckeditor5-ui/src/dialog/dialog.ts

```diff
@@ -115,13 +103,35 @@ export default class Dialog extends Plugin {
 		position,
 		onHide
 	}: DialogDefinition ) {
+		const editor = this.editor;
+
+		Dialog.view = new DialogView( editor.locale, () => {
+			return editor.editing.view.getDomRoot( editor.model.document.selection.anchor!.root.rootName )!;
+		}, () => {
+			return editor.ui.viewportOffset;
+		} );
+
+		Dialog.view.on<DialogViewCloseEvent>( 'close', () => {
+			this.hide();
+		} );
+
+		// If the dialog was closed by opening in another editor instance,
+		// we need to update the state in the previous instance manually.
+		Dialog.view.on( 'destroy', () => {
+			this.id = '';
+			this.isOpen = false;
+		} );
+
+		editor.ui.view.body.add( Dialog.view );
+		editor.ui.focusTracker.add( Dialog.view.element! );
```

Comment:
ATM there are references stored to previous dialog elements in the focusTracker. We should remove elements when views are destroyed.